samples/ProjectGrandmaster.Unity/Assets/Script/Board/BoardInformation.cs

```diff
@@ -0,0 +1,633 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using System;
+using Microsoft.MixedReality.USYD.ChessPiece;
+using UnityEngine.UI;
+using TMPro;
+
+namespace Microsoft.MixedReality.USYD.Board
+{
+    public class BoardInformation : MonoBehaviour
+    {
+        enum Colours { White, Black };
+        Colours turn;
+
+        GameObject[,] board;
+
+        PieceAction pieceAction;
+
+        List<GameObject> piecesOnBoard;
+
+        GameObject blackKing;
+        GameObject whiteKing;
+
+        public LayerMask chessboardLayer;
+
+        // Piece being handled by the user
+        public bool PieceHandling { get; set; }
+
+        // GameObject related to displaying end result
+        public GameObject EndGameResult;
+        public GameObject whiteText;
+        public GameObject blackText;
+        public GameObject leftText;
+        public GameObject rightText;
+
+        // Piece cannot be moved until previous move is complete
+        public bool CanMove { get; set; }
+
+        public bool Check { get; set; }
+        public bool GameEnded { get; set; }
+        public bool AIEnabled { get; set; }
+        public bool ghostActive { get; set; }
+
+        // Side that won. -1 if black, 0 if draw, +1 if white.
+        // Game Ended must also be true
+        public int Winner { get; set; }
+
+        // Arrow on the boardmenu that let's the player know if its their turn
+        // Green if true, else red. 
+        public GameObject whiteArrow;
+        public GameObject blackArrow;
+        public Material playerTurn;
+        public Material opponentTurn;
+
+        // Variables related to pawn promotion
+        public Mesh mesh { get; set; }
+        public bool promoted { get; set; }
+        public bool meshChosen { get; set; }
+        public GameObject pawnPromo;
+
+        public GameObject boardMenuTileText;
+        public GameObject boardMenuTileTextTwo;
+        public Mesh pawnMesh;
+
+        void Start()
+        {
+            /// <summary>
```

Comment:
We generally don't use this format for commenting code lines, only classes, interfaces, enums (their values), structs, and  public members of classes/structs/interfaces.

You could just put a simple comment here, if you wish to have a comment:
	    // 2D chess layout with individual piece GameObject
            board = new GameObject[8, 8];